server/controller/course.controller.js

```diff
@@ -68,45 +69,19 @@ courseController.post(
 courseController.get(
   '/',
   passport.authenticate('jwt', { session: false }),
-  validateCourseId,
-  (req, res) => {
-    validationHandler(req, res, () => {
-      const { _id } = req.query;
-      getPopulatedCourse({ _id }, res);
-    });
-  }
-);
-
-/**
- * GET/
- * Get a course using school and code through params
- */
-courseController.get(
-  '/:school/:code',
-  passport.authenticate('jwt', { session: false }),
-  validateSingleCourse,
-  (req, res) => {
-    validationHandler(req, res, () => {
-      const { code } = req.params;
-      const school = req.params.school.toUpperCase();
-
-      getPopulatedCourse({ school, code }, res);
-    });
-  }
-);
-
-/**
- * GET/
- * Get all courses using school through params
- */
-courseController.get(
-  '/:school',
-  passport.authenticate('jwt', { session: false }),
-  validateSchoolCourses,
+  validateCourseQuery,
   (req, res) => {
-    const school = req.params.school.toUpperCase();
     validationHandler(req, res, () => {
-      getPopulatedCourse({ school }, res);
+      const { _id, school, department, code } = req.query;
+      getPopulatedCourse(
+        {
+          ...(_id && { _id }),
```

Comment:
This looks very nice, where did you learn this?

Reply:
The second to the top answer of this stackoverflow post:
https://stackoverflow.com/questions/11704267/in-javascript-how-to-conditionally-add-a-member-to-an-object